include/boost/compute/random/discrete_distribution.hpp

```diff
@@ -71,52 +88,65 @@ class discrete_distribution
     {
     }
 
-    /// Returns the value of n
-    result_type n() const
-    {
-        return m_n;
-    }
-
     /// Returns the probabilities
     ::std::vector<double> probabilities() const
     {
         return m_probabilities;
     }
 
-    /// Generates uniformily distributed integers and stores
+    /// Returns the minimum potentially generated value.
+    result_type BOOST_PREVENT_MACRO_SUBSTITUTION min() const
```

Comment:
I think the `BOOST_PREVENT_MACRO_SUBSTITUTION` macro needs to go between the function name and the parentheses. Something like:

```
result_type min BOOST_PREVENT_MACRO_SUBSTITUTION () const
```